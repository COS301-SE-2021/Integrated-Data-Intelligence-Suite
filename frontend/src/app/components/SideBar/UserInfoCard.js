import {Button, Card, Dropdown, Menu, Skeleton, Avatar } from "antd";
// import Avatar from "antd/es/avatar/avatar";
import {UpOutlined} from "@ant-design/icons";
import React, {Component} from 'react';
import ExitMenuTooltip from "./ExitMenuTooltip";
const { Meta } = Card;

function setUserName(){
    const localUser = localStorage.getItem("user");
    if(localUser){
        return JSON.parse(localUser);
    }else{
        return "user";
    }
}

class UserInfoCard extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            loading: true,
            collapsed: false,
            user: setUserName()
        };
    }


    onChange = checked => {
        this.setState({loading: !checked});

    };

    onCollapse(collapsed) {
        if (this.state.collapsed) {


        }

    }

    render() {
        const {loading, user} = this.state;

        return (
            <>
                {/*<Switch checked={!loading} onChange={this.onChange} style={{width: 40}}/>*/}
                <Card id={'user_avatar_card'}>
                    <Skeleton loading={false} avatar active>


                        {/*The drop down menu that allows the user to log out or lock the app*/}
                        <Dropdown
                            overlay={<ExitMenuTooltip/>}
                            placement="topCenter"
                            arrow={true}
                            trigger={'click'}
                        >
                            <Button
                                id={'exit_menu_button'}
                                icon={
                                    <UpOutlined
                                        className={'exit_menu_ellipsis_icon'}
                                    />
                                }
                            >
                            </Button>
                        </Dropdown>

                        {/*The user Avatar Image*/}
                        <Meta
                            id={'meta_id'}
                            className={'user_meta_card'}
                            title={user.username}
                            // description={'name@example.com'}
                            // avatar={
                            //     <Avatar
                            //         id={'user_avatar_pic'}
                            //         shape={'round'}
                            //     >
                            //         {user.username.substring(0,1)}
                            //     </Avatar>
                            // }
                        />
                    </Skeleton>
                </Card>
            </>
        );
    }
}

export default UserInfoCard;
