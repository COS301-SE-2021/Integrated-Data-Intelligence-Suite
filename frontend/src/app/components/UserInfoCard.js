import {Button, Card, Dropdown, Menu, Skeleton} from "antd";
import Meta from "antd/es/card/Meta";
import Avatar from "antd/es/avatar/avatar";
import {EllipsisOutlined, LockOutlined, PoweroffOutlined} from "@ant-design/icons";
import React, {Component} from 'react';
import ExitMenuOverlay from "./ExitMenuOverlay";

class UserInfoCard extends React.Component {
    state = {
        loading: true,
    };

    onChange = checked => {
        this.setState({loading: !checked});
    };

    render() {
        const {loading} = this.state;

        return (
            <>
                {/*<Switch checked={!loading} onChange={this.onChange} style={{width: 40}}/>*/}
                <Card id={'user_avatar_card'}>
                    <Skeleton loading={false} avatar active>
                        <Meta
                            id={'meta_id'}
                            className={'user_meta_card'}
                            title="Myron Lopes"
                            avatar={
                                <Avatar id={'user_avatar_pic'}
                                        shape={'round'}>
                                    M
                                </Avatar>
                            }
                        />
                        <Dropdown
                            overlay={<ExitMenuOverlay/>}
                            placement="topCenter"
                            arrow={true}
                            trigger={'click'}
                        >
                            <Button id={'exit_menu_button'}
                                    icon={
                                        <EllipsisOutlined
                                            className={'exit_menu_ellipsis_icon'}
                                            // style={{fontSize: '25px'}}
                                        />
                                    }
                            >
                            </Button>
                        </Dropdown>
                    </Skeleton>
                </Card>
            </>
        );
    }
}

export default UserInfoCard;
