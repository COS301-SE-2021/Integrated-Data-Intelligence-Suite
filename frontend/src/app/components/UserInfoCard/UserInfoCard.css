/*************************************************************/
/********************* User Avatar  Card ***************************/
/************Found in the Bottom Left corner********************/

/* The div that holds the entire user_avatar card*/
#user_avatar_card {
    /*float: right;*/
    /*height: 100%;*/
    width: 90%;
    border: 1px solid transparent;
    padding: 10px;
    background: var(--light-black);
    /*margin-left: 8px;*/
    color: white;
    display: grid;
    justify-items: center;
    justify-self: center;
    align-self: center;
    text-align: center;
    align-items: center;
    align-content: center;
    border-radius: 6px;
    margin: auto;
}

#user_avatar_card > .ant-card-body {
    padding: 0;
    border: 0;
    border-radius: 6px;
    background: transparent;


    /*border: 3px solid black;*/
    /*display: grid;*/
    /*grid-template-columns: repeat(2, min-content);*/
    /*grid-template-rows: 100%;*/
}

#user_avatar_pic {
    /*border: 2px solid black;*/
    color: white;
    background: orange;
    float: right;
    width: 39px;
    height: 39px;
    line-height: 1.5715;
    font-size: 16px;
    border-radius: 100%;
    margin-left: 4px;
}

#meta_id {
    /*border: 2px solid purple;*/
    /*border: 0;*/
    float: right;
    text-transform: capitalize;
    font-weight: 600;
    /*width: 78%;*/
    /*height: 100%;*/
    background: #04aa6d00;


    grid-column-start: 1;
    grid-column-end: 2;

}

.ant-card-meta {
    /*border: 2px solid blue;*/
    margin: 0;
    background: transparent;
    border: 0;

}

/*The div wrapping the users avatar*/
#meta_id > .ant-card-meta-avatar {
    float: right;
    padding-right: 0;
    padding-left: 0;
    margin-top: 18px;
    background: transparent;
    border: 0;
}

/*The div for the users Name and a description*/
#meta_id > .ant-card-meta-detail {
    float: left;
    justify-content: left;
    /*margin-top: 19px;*/
    background: transparent;
    border: 0;
}

/*The users' Name
* AKA the title in the JS code
*/
#meta_id > .ant-card-meta-detail > .ant-card-meta-title {
    color: white;
    border: 0;
    font-size: 13px;
    font-weight: 500;
    margin-bottom: 0;
    text-overflow: ellipsis;
    /*margin-top: 25px;*/
    /*background: black;*/

}

/*The grey out section underneath the name*/
#meta_id > .ant-card-meta-detail > .ant-card-meta-description {
    font-size: 11px;
    height: 17px;
    width: 12px;
    color: #9c9494;
    background: transparent;
}

/*The DROP Down button*/
#exit_menu_button {
    /*border: 1px solid green;*/
    border: 0;
    box-shadow: none;
    font-size: 0;
    float: right;
    /*transform: rotate(*/
    /*        45deg*/
    /*);*/
    /*margin-top: 24px;*/
    width: 31px;
    height: 23px;
    padding: 0;
    background: transparent;
    color: black;

    /*grid-column-start: 2;*/
    /*grid-column-end: 3;*/
}

/*The actual Image Icon
* Currently its a ">" icon facing downwards i.e a down error
*/
#exit_menu_button > .exit_menu_ellipsis_icon > svg {
    color: black;
    background: transparent;
}

/*When you hover over the Downward arrow icon change color*/
#exit_menu_button > .exit_menu_ellipsis_icon > svg:hover {
    color: var(--pink);
    background: transparent;
}

/*When you Click on the Downward arrow icon change color*/
button#exit_menu_button.ant-btn.ant-btn-icon-only.ant-dropdown-trigger.ant-dropdown-open >
span.anticon.anticon-caret-up.exit_menu_ellipsis_icon > svg {
    color: #000000;
    background: transparent;
}





