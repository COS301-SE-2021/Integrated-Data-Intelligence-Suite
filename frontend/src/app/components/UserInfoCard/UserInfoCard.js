import {Button, Card, Dropdown, Menu, Skeleton, Avatar} from "antd";
// import Avatar from "antd/es/avatar/avatar";
import {UpOutlined, CaretUpFilled} from "@ant-design/icons";
import React, {Component} from 'react';
import ExitMenuTooltip from "../ExitMenuTooltip/ExitMenuTooltip";
import ExitMenuDropDown from "../ExitMenuDropDown/ExitMenuDropDown";

const {Meta} = Card;

function setUserName() {
    const localUser = localStorage.getItem("user");
    if (localUser) {
        return JSON.parse(localUser);
    } else {
        return "user";
    }
}

class UserInfoCard extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            user: setUserName()
        };
    }


    render() {
        const {loading, user} = this.state;

        return (
            <>
                <Card id={'user_avatar_card'}>
                    <Skeleton loading={false} avatar active>

                        {/*The drop down menu that allows the user to log out or lock the app*/}
                        <ExitMenuDropDown/>

                        {/*The user Avatar Image*/}
                        <Meta
                            id={'meta_id'}
                            className={'user_meta_card'}
                            title={user.username}
                            // description={'name@example.com'}
                        />
                    </Skeleton>
                </Card>
            </>
        );
    }
}

export default UserInfoCard;
