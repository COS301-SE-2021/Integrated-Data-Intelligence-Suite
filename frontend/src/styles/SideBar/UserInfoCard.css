/*************************************************************/
/********************* User Avatar  Card ***************************/
/************Found in the Bottom Left corner********************/

/* The div that holds the entire user_avatar card*/
#user_avatar_card {
    /*bottom: -50vh;*/
    bottom:0;
    height: 10%;
    /* color: black; */
    /*background: #15B761;*/
    background: transparent;
    /*border: 0;*/
    border: 2px solid yellow;
    /*margin-left: 8px;*/
    /*margin-right: 8px;*/
    /*border-radius: 6px;*/

}

#user_avatar_card > .ant-card-body {
    padding: 0;
    border: 0;
    border-radius: 6px;
    background: transparent;
}

#user_avatar_pic {
    /*border: 2px solid black;*/
    color: white;
    background: orange;
    float: right;
    width: 39px;
    height: 39px;
    line-height: 1.5715;
    font-size: 16px;
    border-radius: 100%;
    margin-left: 4px;
}

#meta_id {
     border: 2px solid purple;
    /*border: 0;*/
    float: right;
    /*width: 78%;*/
    /*height: 100%;*/
    background: #04aa6d00;

}

.ant-card-meta {
    /*border: 2px solid blue;*/
    margin: 0;
    background: transparent;
    border: 0;

}

/*The div wrapping the users avatar*/
#meta_id > .ant-card-meta-avatar {
    float: left;
    padding-right: 0;
    padding-left: 0;
    margin-top: 18px;
    background: transparent;
    border: 0;
}

/*The div for the users Name and a description*/
#meta_id > .ant-card-meta-detail {
    float: left;
    justify-content: left;
    /*margin-top: 19px;*/
    background: transparent;
    border: 0;

}

/*The users' Name*/
#meta_id > .ant-card-meta-detail > .ant-card-meta-title {
    color: white;
    border: 0;
    font-size: 13px;
    font-weight: 600;
    margin-bottom: 0;
    margin-top: 25px;
    background: transparent;
    border: 0;

}

/*The grey out section underneath the name*/
#meta_id > .ant-card-meta-detail > .ant-card-meta-description {
    font-size: 11px;
    height: 17px;
    width: 12px;
    color: #9c9494;
    background: transparent;
}

/*The ellipsis button*/
#exit_menu_button {
    /*border: 1px solid green;*/
    border: 0;
    font-size: 0;
    float: right;
    /*transform: rotate(*/
    /*        90deg*/
    /*);*/
    margin-top: 24px;
    width: 31px;
    height: 23px;
    padding: 0;
    background: transparent;
}

#exit_menu_button > .exit_menu_ellipsis_icon > svg {
    color: white;
    background: transparent;
}


