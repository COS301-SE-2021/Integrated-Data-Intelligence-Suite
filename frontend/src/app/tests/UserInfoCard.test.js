import React from 'react';
import renderer from 'react-test-renderer';
import UserInfoCard from "../components/UserInfoCard/UserInfoCard";
import {unmountComponentAtNode} from "react-dom";


let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});


it('renders correctly', () => {
    const tree = renderer
        .create(<UserInfoCard/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
