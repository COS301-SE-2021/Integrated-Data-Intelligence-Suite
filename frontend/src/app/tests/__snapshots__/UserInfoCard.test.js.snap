// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ant-card ant-card-bordered"
  id="user_avatar_card"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <button
      className="ant-btn ant-btn-icon-only ant-dropdown-trigger"
      id="exit_menu_button"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <span
        aria-label="up"
        className="anticon anticon-up exit_menu_ellipsis_icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="up"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
          />
        </svg>
      </span>
    </button>
    <div
      className="ant-card-meta user_meta_card"
      id="meta_id"
    />
  </div>
</div>
`;
